packages/core/src/components/tag-input/resizeableInput.tsx

```diff
@@ -0,0 +1,50 @@
+/* !
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ */
+
+import React, { forwardRef, useEffect, useRef, useState } from "react";
+
+import { HTMLInputProps } from "../../common";
+
+type ResizeableInputProps = HTMLInputProps;
+export type Ref = HTMLInputElement;
+
+export const ResizeableInput = forwardRef<Ref, ResizeableInputProps>(function ResizeableInput(props, ref) {
+    const [content, setContent] = useState("");
+    const [width, setWidth] = useState(0);
+    const span = useRef<HTMLSpanElement>(null);
+
+    useEffect(() => {
+        if (span.current) {
```

Comment:
nit: please use explicit boolean coercion

```suggestion
        if (span.current != null) {
```